packages/babel-core/src/config/config-chain.js

```diff
@@ -351,6 +472,42 @@ function dedupDescriptors(
   }, []);
 }
 
+function configIsApplicable(
+  { options }: OptionsAndDescriptors,
+  dirname: string,
+  context: ConfigContext,
+): boolean {
+  return (
+    (options.test === undefined ||
+      configFieldIsApplicable(context, options.test, dirname)) &&
+    (options.include === undefined ||
+      configFieldIsApplicable(context, options.include, dirname)) &&
```

Comment:
What is the difference between `test` and `include`?

Reply:
I think it's the same actually just matching webpack

Reply:
Old webpack docs are gone, but I believe it used to be that `test` had to be a regex (e.g. `/\.js$/`), while `include`/`exclude` had to be a string (e.g. `srcDir`)? In any case, they have gained the connotations of "file type" and "directories".